### ISSUE TITLE
Add algorithmic choice as a (separably) tuneable parameter

### ISSUE THREAD

**Issue by tmcdonell** [vollmerm/shallow-fission#14]
Provide a mechanism where an algorithm can provide a plurality of implementations and the auto-tuner selects between them. This is separate from the choice of how to split up input data.

Should be able to leverage the composable autotuning framework of FHPC'15.

**Comment by tmcdonell** [comment]
Divide-and-conquer matrix-multiply is an example.

**Comment by vollmerm** [comment]
Where does this fit in to the story of the paper as we're telling it now? We're focusing on the whole "programmer doesn't need to do anything and fissioning happens" angle. Doing petabricks-style first class algorithmic choices with this early embedding thing is clearly a good idea, but it seems tangential to the paper as it's written now.

One option would be to use the next few days to double down on auto tuning and try to do that very thoroughly, and focus less on the clever Haskell stuff. This kinda makes sense since the clever Haskell stuff isn't done, but that's changing the whole paper with only a few days to go.

**Comment by tmcdonell** [comment]
I agree, this is tangential to what is written now.

I'm really just going and adding ideas of "what would make this into a paper", rather than "what can we add in the next few days" because, honestly, I'm not convinced that what little we have (even if it all worked) is worthy of submission.

**Comment by vollmerm** [comment]
I agree. After all this work we don't really have anything terribly impressive. I guess that happens sometimes.

**Comment by tmcdonell** [comment]
Yeah, not having impressive results is another nail in the coffin.
